src/atoms/Button/__tests__/Button.native.test.js

```diff
@@ -1,11 +1,139 @@
+  });
+
+  it('Renders a Secondary <Button />', () => {
+    const { container, getByText } = renderWithTheme(<Button type="secondary">Click Me</Button>);
+    expect(getByText(/click me/i)).toBeTruthy();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('Renders a small Secondary <Button />', () => {
+    const { container, getByText } = renderWithTheme(
+      <Button type="secondary" size="small">
+        Click Me
+      </Button>,
+    );
+    expect(getByText(/click me/i)).toBeTruthy();
     expect(container).toMatchSnapshot();
   });
+
+  it('Renders a large Secondary <Button />', () => {
+    const { container, getByText } = renderWithTheme(
+      <Button type="secondary" size="large">
+        Click Me
+      </Button>,
+    );
+    expect(getByText(/click me/i)).toBeTruthy();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('Renders a disabled Secondary <Button />', () => {
+    const { container, getByText } = renderWithTheme(
+      <Button disabled type="secondary">
+        Click Me
+      </Button>,
+    );
+    expect(getByText(/click me/i)).toBeTruthy();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('Renders a Tertiary <Button />', () => {
+    const { container, getByText } = renderWithTheme(<Button type="tertiary">Click Me</Button>);
+    expect(getByText(/click me/i)).toBeTruthy();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('Renders a small Tertiary <Button />', () => {
+    const { container, getByText } = renderWithTheme(
+      <Button type="tertiary" size="small">
+        Click Me
+      </Button>,
+    );
+    expect(getByText(/click me/i)).toBeTruthy();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('Renders a large Tertiary <Button /> correctly', () => {
+    const { container, getByText } = renderWithTheme(
+      <Button type="tertiary" size="large">
+        Click Me
+      </Button>,
+    );
+    expect(getByText(/click me/i)).toBeTruthy();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('Renders a disabled Tertiary <Button />', () => {
+    const { container, getByText } = renderWithTheme(
+      <Button disabled type="tertiary">
+        Click Me
+      </Button>,
+    );
+    expect(getByText(/click me/i)).toBeTruthy();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('Checks onClick is called for <Button />', () => {
+    const mockPress = jest.fn();
+    const { getByText } = renderWithTheme(<Button onClick={mockPress}>Click Me</Button>);
+    const button = getByText(/click me/i).parentNode;
+    fireEvent.press(button);
+    expect(mockPress).toBeCalledTimes(1);
+  });
+
+  it('Checks onClick is not called for disabled <Button />', () => {
+    const mockPress = jest.fn();
+    const { getByText } = renderWithTheme(
+      <Button disabled onClick={mockPress}>
+        Click Me
+      </Button>,
+    );
+    const button = getByText(/click me/i).parentNode;
+    fireEvent.press(button);
+    expect(mockPress).not.toBeCalled();
+  });
```

Comment:
Can you add a snapshot for each variant of buttton before & after button PressIn. You can refer to Link component for this

Reply:
It won't make a difference coz I am not storing any state/rerendering component, native `underlayColor` is being used. You will always see the `underlayColor` before and after pressIn snapshots.